### ISSUE TITLE
Mage teardown is ignoring some custom CloudWatch alarms

### ISSUE THREAD

**Issue by sagium** [panther-labs/panther#918]
### Describe the bug

After running `mage teardown` we end up with some non-deleted alarms originating from `internal/core/custom_resources/resources/alarms_appsync.go`.

### Steps to reproduce

1. Run `mage teardown`
2. Open your https://console.aws.amazon.com/cloudwatch
3. There should be some alarms living under the "AWS/AppSync" namespace

### Expected behavior

Those alarms should have been deleted along with the rest of the resources upon teardown.

### Environment

Checked out commit when testing bc93a99677b2871b89a3fef1097eb9e976e30610
But this is not something broken, this is just something missing.

### Proposed solution

You could add a function to cleanup those in `tools/mage/teardown.go`, since they don't depend on CloudFormation. Something like that..

```
// Destroy any leftover CloudWatch alarms
// Why use a generic "AWS/AppSync" and not include "Panther" in the namespace?
func destroyCustomAlarms(awsSession *session.Session){
	logger.Debug("checking for leftover Panther CloudWatch alarms")
	client := cloudwatch.New(awsSession)

	// Traversing through all alarms might be slow, maybe find a way to get alarms per Namespace.
	// If the AWS User is made for Panther only, this should work fine as is.
	resp, _ := client.DescribeAlarms(nil)
	deletedAlarmsCounter := 0
	errCount := 0

	for _, alarm := range resp.MetricAlarms {
		alarmNames := []*string{alarm.AlarmName}

		if *alarm.Namespace == "AWS/AppSync" && strings.Contains(strings.ToLower(*alarm.AlarmName), "panther") {
			input := &cloudwatch.DeleteAlarmsInput{AlarmNames: alarmNames}
			if _, err := client.DeleteAlarms(input); err != nil {
				logger.Fatalf("failed to delete alarm: %s %v", *alarm.AlarmName, err)
				errCount++
			}

			deletedAlarmsCounter++
		}
	}

	if deletedAlarmsCounter > 0 {
		logger.Infof("deleted %d CloudWatch alarms", deletedAlarmsCounter)
	}

	if errCount > 0 {
		logger.Fatalf("%d CloudWatch alarms failed to delete", errCount)
	}
}
```

**Comment by austinbyers** [comment]
Thanks for the report! I'll look into it this week

**Comment by austinbyers** [comment]
@sagium this will be fixed in #925

We are migrating to custom resources for all CloudWatch alarms, but the custom resource already includes logic to delete the alarm when CloudFormation deletes the stack. All other custom alarms deleted correctly, we shouldn't need any special teardown code.

It turns out we were using the wrong alarm name, that's all. Good find!

As for this question:

> // Why use a generic "AWS/AppSync" and not include "Panther" in the namespace?

The "namespace" is defined by the metric itself. We don't control that - if you want to use the AppSync metrics, you have to specify the `AWS/AppSync` namespace to get to them:

<img width="406" alt="Screen Shot 2020-05-20 at 1 42 54 PM" src="https://user-images.githubusercontent.com/3608925/82495417-ede66b80-9a9f-11ea-9dca-cd60a6bf2997.png">
